scripts/setup/nrfconnect/get_zap.py

```diff
@@ -0,0 +1,131 @@
+    except:
+        raise RuntimeError(
+            "Encountered problem when trying to read " + zap_version_file + " file.")
+
+def get_zap_current_version():
+    try:
+        cmd_out = subprocess.check_output(['zap', '--version'])
+        cmd_decoded_out = cmd_out.decode('ascii').strip()
+        version = re.findall(r'Version: (.*)\s', cmd_decoded_out)
+        
+        if (len(version) == 0):
+            raise RuntimeError("Couldn't find pattern matching Version:.")
+        elif (len(version) > 1):
+            raise RuntimeError("Found multiple patterns matching Version:.")
+            
+        return version[0]
+    except:
+        return None
+
+def download_recommended_zap_package(version, package_name, location):    
+    try:
+        print("Trying to download ZAP tool package matching your system and recommended version.")
+        response = wget.download("https://github.com/project-chip/zap/releases/download/" + version + "/" + package_name, out=location)
+        print("\n")
+    except:
+        raise RuntimeError("Invalid URL to download ZAP tool package")
+
+def unzip_zap_package(location, filename):
+    try:
+        zip = ZipFile(location + filename + '.zip')
+        zip.extractall(location + filename)
+        zip.close()
+
+    except:
+        raise RuntimeError("Encountered problem when trying to unzip the ZAP tool package.")
+
+def install_zap_package(version, location):
+    current_os = platform.system()
+    if (current_os) == 'Linux':
+        filename = 'zap-linux'
+    elif (current_os) == 'Windows':
+        filename = 'zap-windows'
+    elif (current_os) == 'Darwin':
+        filename = 'zap-mac'
+    else:
+        raise RuntimeError("Couldn't find the proper ZAP tool package for the currently used operating system: " + current_os)
+    
+    download_recommended_zap_package(version, filename + '.zip', location)
+    unzip_zap_package(location, filename)
+    print("ZAP tool package was downloaded and extracted in the given location.")
+    instruction_message = "# Please add the following location to the system PATH: " + location + filename + "/ #"
+    print(f"\33[33m{len(instruction_message)*'#'}\x1b[0m")
+    print(f"\33[33m{instruction_message}\x1b[0m")
+    print(f"\33[33m{len(instruction_message)*'#'}\x1b[0m")
+    
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Script helping to download the ZAP tool in the currently recommended revision.')
+    parser.add_argument(
+        "-l", "--location", help="Path to the location that should be used for storing ZAP tool package.", type=str, required=True)
+    args = parser.parse_args()
+
+    location = args.location
+    if(location[-1] != '/'):
+        location = location + '/'
+
+    zap_recommended_version = get_zap_recommended_version()
+    # Assuming format vYYYY.MM.DD-nightly, extract only date without v prefix and potential suffix
+    zap_recommended_version_number = zap_recommended_version[1:11]
```

Comment:
Extracting only a specific number of bytes can be dangerous here because something can change.

Reply:
But do we have any other option than assuming some pattern?

Reply:
If we know a constant variable name for e.g. `ZAP_VERSION=`vYYYY.MM.DD-nightly we can read that line from the file and then use:
line = ZAP_VERSION=vYYYY.MM.DD-nightly
position = line.find("ZAP_VERSION=")

and finally:
zap_recommended_version_number = line[position:]

Reply:
Of course, it can be simplified, but I wanted to show my thinking.

Reply:
Discussed F2F